policybot/pkg/storage/spanner/read.go

```diff
@@ -276,6 +277,47 @@ func (s store) ReadBotActivityByID(context context.Context, orgID string, repoID
 	return &result, nil
 }
 
+func (s store) ReadTestResultByName(context context.Context, orgID string,
+	repoID string, testName string, prNum int64, runNum int64) (*storage.TestResult, error) {
+	row, err := s.client.Single().ReadRow(context, testResultTable, testResultKey(orgID, repoID, testName, prNum, runNum), testResultColumns)
+	if spanner.ErrCode(err) == codes.NotFound {
+		return nil, nil
+	} else if err != nil {
+		return nil, err
+	}
+	var result storage.TestResult
+	if err := row.ToStruct(&result); err != nil {
+		return nil, err
+	}
+
+	return &result, nil
+}
+
+/*
```

Comment:
Following the pattern in this package:

- This should be QueryXXX since it returns more than one item.

- And again for consistency, please make this use a callback function which gets invoked once per item, instead of returning a slice. This makes the actual query able to produce incremental results, rather than blocking until all results are provided. If you need to, the caller can always accumulate into a slice if it wants to.